NBitcoin.Altcoins/Monacoin.cs

```diff
@@ -0,0 +1,220 @@
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using NBitcoin.Protocol;
+using NBitcoin.RPC;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+
+namespace NBitcoin.Altcoins
+{
+	public class Monacoin : NetworkSetBase
+	{
+		public static Monacoin Instance { get; } = new Monacoin();
+
+		public override string CryptoCode => "MONA";
+
+		private Monacoin()
+		{
+
+		}
+		//Format visual studio
+		//{({.*?}), (.*?)}
+		//Tuple.Create(new byte[]$1, $2)
+		static Tuple<byte[], int>[] pnSeed6_main = {
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x31,0xd4,0xa6,0xb5}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x68,0x9c,0xee,0xcb}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x68,0xe9,0x7a,0xa9}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x71,0x92,0x44,0xfb}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x7c,0x27,0x04,0x93}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x80,0xc7,0xd6,0xa8}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x80,0xc7,0xfe,0xd8}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x90,0x4c,0x03,0x88}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x99,0x78,0x27,0x59}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xc7,0x7f,0x6c,0xa1}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xca,0xb5,0x65,0xcd}, 9401),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xdb,0x75,0xf8,0x37}, 9401),
+};
+		static Tuple<byte[], int>[] pnSeed6_test = {
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x6f,0x67,0x3b,0x7d}, 19403),
+	Tuple.Create(new byte[]{0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0x99,0x78,0x27,0x59}, 19403)
+};
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		public class MonacoinConsensusFactory : ConsensusFactory
+		{
+			private MonacoinConsensusFactory()
+			{
+			}
+
+			public static MonacoinConsensusFactory Instance { get; } = new MonacoinConsensusFactory();
+
+			public override BlockHeader CreateBlockHeader()
+			{
+				return new MonacoinBlockHeader();
+			}
+			public override Block CreateBlock()
+			{
+				return new MonacoinBlock(new MonacoinBlockHeader());
+			}
+		}
+
+		public class MonacoinBlockHeader : BlockHeader
+		{
+			public override uint256 GetPoWHash()
+			{
+				//TODO: Fix code in this function.
```

Comment:
The last issue for this pull request is caused by here. Monacoin was hard-forked before.